convert.sh

```diff
@@ -1,74 +1,79 @@
-    -i $arg1           `#input file: $arg1` \
-    -f sbc             `#output format: SBC` \
-    -ar 16000          `# audio rate: 16 kHz` \
-    -ac 1              `# audio channel: #1` \
-    -aq 16             `# audio quality: 16 (for SBC this means bitpool=16)` \
-    -map_metadata -1   `# ????` \
-    out-$hash.sbc            `# output to out.raw`
-
-  # sbcinfo out-$hash.sbc
-  xxd -i out-$hash.sbc            `# output in C include file style` \
+    -v quiet                  `# verbosity - other options are "quiet", "error", "panic"` \
+    -i "$1"                   `#input file: $1` \
+    -f sbc                    `#output format: SBC` \
+    -ar 16000                 `# audio rate: 16 kHz` \
+    -ac 1                     `# audio channel: #1` \
+    -aq 16                    `# audio quality: 16 (for SBC this means bitpool=16)` \
+    -map_metadata -1          `# ????` \
+    "$filename"               `# output to out.raw`
+
+  xxd -i "$filename"          `# output in C include file style` \
     | head -n -2              `# skip last two xxd outline lines (skip C formatting)` \
     | tail -n +2              `# start output on line 2 of xxd output (skip C formatting)` \
     | sed 's/ //g'            `# remove spaces` \
     | tr --delete '\n'        `# remove newlines` \
     | sed 's/,/\,\n/16; P; D' `# collect into lines with the right length` \
-    > $arg2
-  rm out-$hash.sbc
+    > "$2"
+
+  rm "$filename"
 }
 
 audio_to_cpp() {
-  hash=$(echo $arg1 | md5sum | cut -d" " -f1)
-  filename="out-$hash.sbc"
-  varname=$(echo $arg1 | rev | cut -c5- | cut -d"/" -f1,2 | rev | tr '[:lower:]' '[:upper:]' | tr "/" _)
+  varname="$(echo "$1" | rev | cut -f 2- -d '.' | cut -d '/' -f1,2 | rev | tr '[:lower:]/' '[:upper:]_')"
+
   ffmpeg -y \
-    -v quiet            `# verbosity - other options are "quiet", "error", "panic"` \
-    -i $arg1           `#input file: $arg1` \
-    -f sbc             `#output format: SBC` \
-    -ar 16000          `# audio rate: 16 kHz` \
-    -ac 1              `# audio channel: #1` \
-    -aq 16             `# audio quality: 16 (for SBC this means bitpool=16)` \
-    -map_metadata -1   `# ????` \
-    $varname            `# output to something like EN_SOUND_POWER_ON`
-
-  echo "#include <stdint.h>\n" > $arg2
-
-  # sbcinfo out-$hash.sbc
-  xxd -i $varname            `# output in C include file style` \
-    >> $arg2
-
-  sed -i "s/unsigned char/uint8_t/g" $arg2
-  
-  #rm $filename
+    -v quiet         `# verbosity - other options are "quiet", "error", "panic"` \
+    -i "$1"          `#input file: $1` \
+    -f sbc           `#output format: SBC` \
+    -ar 16000        `# audio rate: 16 kHz` \
+    -ac 1            `# audio channel: #1` \
+    -aq 16           `# audio quality: 16 (for SBC this means bitpool=16)` \
+    -map_metadata -1 `# ????` \
+    "$varname"       `# output to something like EN_SOUND_POWER_ON`
+
+  printf '#include <stdint.h>\n\n' > "$2"
```

Comment:
Does `printf` give an advantage over `echo` here? I typically just use echo so I'm not sure what the real difference between the old and new lines is here.

Reply:
`echo` doesn't necessarily support escape sequences such as `\n`, according to the POSIX spec, whereas `printf` does. If you're not sure which one to use, `echo` is usually fine, but `printf` is better for more intricate things, such as repeating strings, etc.